nixos/modules/services/cluster/dynomite.nix

```diff
@@ -0,0 +1,59 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
```

Comment:
Maybe you could add something like this to your `let`:
```
redisLocal = any (value: value == "localhost:${config.services.redis.port}") cfg.settings.servers;
```

And then you could conditionally enable `redis` as well as conditionally add it as a dependency to your systemd unit.

On a side note I might personally be inclined to add `servers` as a separate nixos option because it is so central to what this service does... but you certainly don't need to do that, and that option could be added at a later time if found to be valuable.